app/controllers/users/passwords_controller.rb

```diff
@@ -2,19 +2,29 @@ module Users
   class PasswordsController < Devise::PasswordsController
     include ValidEmailParameter
 
-    before_action :confirm_valid_token, only: [:edit]
-
     def create
-      RequestPasswordReset.new(params[:user][:email]).perform
+      email = params[:user][:email]
+
+      RequestPasswordReset.new(email).perform
+
+      analytics_user = User.find_by_email(email) || AnonymousUser.new
```

Comment:
I mostly agree.

I guess it depends weather we want to log an attempt to reset the password for a non-existant user as completely different type of of event.

```
if user.present?
   "Password reset requested for $user.role"
else
   "Attempt to reset password for non-existent user"
```

Also, looking the current message don't you think it should say "password reset" rather than just "reset"?